src/Domain/Creators/Cylinder.cpp

```diff
@@ -43,7 +47,27 @@ Cylinder::Cylinder(
           std::move(initial_number_of_grid_points)),         // NOLINT
       use_equiangular_map_(use_equiangular_map),             // NOLINT
       radial_partitioning_(std::move(radial_partitioning)),  // NOLINT
-      height_partitioning_(std::move(height_partitioning)) {}
+      height_partitioning_(std::move(height_partitioning)),
+      boundary_condition_(std::move(boundary_condition)) {
+  if (not radial_partitioning_.empty() and boundary_condition_ != nullptr) {
+    PARSE_ERROR(
+        context,
+        "Do not support specifying boundary conditions and multiple radial "
+        "partitionings. Support can be added if desired.");
+  }
+  if (not height_partitioning_.empty() and boundary_condition_ != nullptr) {
+    PARSE_ERROR(
+        context,
+        "Do not support specifying boundary conditions and multiple radial "
+        "partitionings. The domain creator code to support this is written but "
+        "untested. To enable, please add tests.");
+  }
+  using domain::BoundaryConditions::is_periodic;
+  if (is_periodic(boundary_condition_)) {
+    PARSE_ERROR(context,
+                "Cannot have periodic boundary conditions with a cylinder");
```

Comment:
reword to something like "Periodic boundary conditions are not supported yet for a cylinder. You can add support for periodic boundary conditions along the symmetry axis to the Cylinder domain creator."

Reply:
Periodic in that direction is already support...